src/vm/builtins/builtin_runner/keccak.zig

```diff
@@ -265,53 +265,42 @@ pub const KeccakBuiltinRunner = struct {
     /// Calculates the Keccak hash of the input message.
     ///
     /// This function computes the Keccak hash of the provided input message and returns
-    /// it as an `ArrayList(u8)`. The Keccak hash function involves multiple steps of data
+    /// it as an fixed array based on block type. The Keccak hash function involves multiple steps of data
     /// processing.
     ///
     /// # Arguments
     ///
-    /// - `allocator`: An allocator for managing memory.
     /// - `input_message`: A pointer to the input message as an array of bytes.
     ///
     /// # Returns
     ///
-    /// An `ArrayList(u8)` containing the Keccak hash.
-    fn keccakF(allocator: Allocator, input_message: *[]const u8) !ArrayList(u8) {
-        var result = ArrayList(u8).init(allocator);
-        var vec = ArrayList(u64).init(allocator);
-        defer vec.deinit();
+    /// An fixed array based on block type, containing the Keccak hash.
+    /// current block type hardcoded to u64.
+    fn keccakF(input_message: *[]const u8) ![KeccakPrimitives.PLEN * @sizeOf(u64)]u8 {
+        var result = [_]u8{0} ** (KeccakPrimitives.PLEN * @sizeOf(u64));
+
+        // 1600 bits = 200 bytes = 25 u64
+        var st: std.crypto.core.keccak.KeccakF(@bitSizeOf(u64) * KeccakPrimitives.PLEN) = .{
```

Comment:
Maybe we can find another name for this variable, in order to avoid `st.st` usage after that, wdt?

Reply:
st -> hashState?

Reply:
Yes sounds good